File: PTA/11.1课后题/7-1 查找书籍/main.fs
open System
let count = Convert.ToInt32(Console.ReadLine())
type book = {name:string;price:float}
let mutable max = {name="";price=0.0}
let mutable min = {name="";price=1000000.0}
for i = 1 to count do
    let input = {name=Console.ReadLine();price=Convert.ToDouble(Console.ReadLine())}
    if input.price>max.price then
        max<-input
    if input.price<min.price then 
        min<-input
Console.Write(String.Format("{0:F}",max.price)+", "+max.name+"\n"+String.Format("{0:F}",min.price)+", "+min.name)

File: PTA/11.1课后题/7-1 查找书籍/main.cs
using static System.Console;
using static System.Convert;
using System.Collections.Generic;
using System.Linq;
using System;
namespace TEMP_CS
{
    public class Book
    {
        private string name;
        private double price;
        public Book(string name,double price)
        {
            this.name = name;
            this.price = price;
        }
        public string Name { get => name;}
        public double Price { get => price;}
        public bool Display()
        {
            WriteLine($"{String.Format("{0:F}", Price)}, {Name}");
            return true;
        }
    }
    public class Program
    {
        public static void Main()
        {
            List<Book> books = new List<Book>();
            int count = ToInt32(ReadLine());
            for(int i=0;i<count;++i)
            {
                books.Add(new Book(ReadLine(), ToDouble(ReadLine())));
            }
            books.OrderBy(book => book.Price).Reverse().ToList()[0].Display();
            books.OrderBy(book => book.Price).ToList()[0].Display();
        }
    }
}
